lib/ui/painting.dart

```diff
@@ -1830,6 +1835,49 @@ class Image {
   String toString() => _image.toString();
 }
 
+class ImagePerfData {
+  ImagePerfData(this.resourceDecodeTime);
+  int resourceDecodeTime;
+}
+
+// ignore: avoid_classes_with_only_static_members
+class ImagePerfInfos {
+  static Expando<ImagePerfData> expando = Expando<ImagePerfData>('ImagePerfInfos');
+  static int getResourceDecodeTime(Image image) {
+    final ImagePerfData? info = expando[image];
+    if (info == null) {
+      return 0;
+    }
+    return info.resourceDecodeTime;
+  }
+  static void setResourceDecodeTime(Image image, int time) {
+    expando[image] = ImagePerfData(time);
+  }
+  static void setResourceDecodeTimePri(_Image image, int time) {
+    if (!kReleaseMode) {
+      image.resourceDecodeTime = time;
+      for(final Image imageh in image.handles) {
+        setResourceDecodeTime(imageh, time);
+      }
+    }
+  }
+}
+
+extension ImagePerfInfo on Image {
```

Comment:
Style guide says we should avoid this: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension